MapboxDirections/MBVisualInstruction.swift

```diff
@@ -63,19 +82,25 @@ open class VisualInstruction: NSObject, NSSecureCoding {
             }
         }
         
-        self.init(distanceAlongStep: distanceAlongStep, primaryText: primaryText, primaryTextComponents: primaryTextComponents, secondaryText: secondaryText, secondaryTextComponents: secondaryTextComponents)
+        let maneuverType = ManeuverType(description: json["type"] as! String) ?? .none
+        let maneuverDirection = ManeuverDirection(description: json["modifier"] as! String)  ?? .none
```

Comment:
Should we unwrap the modifier gracefully or make it non-optional server-side?

Reply:
This is something we have sort of struggled with deciding upon recently. It is a required key on the server, but as you have seen, there is a bug which omits it in some cases causing a crash.

Not really sure what the right answer here.

/cc @1ec5 @JThramer @lyzidiamond

Reply:
🤔 hmm, this is a stumper. If a missing `maneuverType` / `maneuverDirection` leads to significant undefined behavior later-on, then silently failing would be a bad idea. 

I think a chore switching all of the model logic over to `Codable` will help this issue out _a lot_ (as `Codable` will throw an error when it fails, which you can then turn into a understandable user error message), but if we want to do a `preconditionFailure(message:)` or something for now that spells out to the user exactly what is going on, that would be cool.

Reply:
A client library should never silently swallow bugs that originate from the server. I’m OK with an assertion (or `preconditionFailure`) in this case for now, but ultimately an exception (that leads to the error handler being called) would be the ideal way to handle malformed server responses.